tests/integration/conftest.py

```diff
@@ -54,17 +70,80 @@ async def unit_ip_list(get_unit_ip_list):
 
 
 @pytest_asyncio.fixture(scope="module")
-async def app(ops_test: pytest_operator.plugin.OpsTest, content_cache_image: str):
+async def nginx_integrator_app(ops_test: OpsTest):
+    nginx_integrator_app_name = "nginx-ingress-integrator"
+    nginx_integrator_app = await ops_test.model.deploy(nginx_integrator_app_name)
+    await ops_test.model.wait_for_idle()
+    assert (
+        ops_test.model.applications[nginx_integrator_app_name].units[0].workload_status
+        == ActiveStatus.name
+    )
+    yield nginx_integrator_app
+
+
+@pytest_asyncio.fixture(scope="module")
+async def app(
+    ops_test: OpsTest, app_name: str, content_cache_image: str, nginx_integrator_app: Application
+):
     """Content-cache-k8s charm used for integration testing.
 
-    Builds the charm and deploys it for testing purposes.
+    Deploy kubecon charm, builds the charm and deploys it for testing purposes.
     """
-    my_charm = await ops_test.build_charm(".")
+    hello_kubecon_app_name = "hello-kubecon"
+    ops_test.model.deploy(hello_kubecon_app_name)
+    await ops_test.model.deploy(hello_kubecon_app_name)
+    app_charm = await ops_test.build_charm(".")
     application = await ops_test.model.deploy(
-        my_charm,
+        app_charm,
+        application_name=app_name,
         resources={"content-cache-image": content_cache_image},
         series="jammy",
     )
     await ops_test.model.wait_for_idle()
 
+    apps = [app_name, nginx_integrator_app.name, hello_kubecon_app_name]
+    await ops_test.model.add_relation(hello_kubecon_app_name, f"{app_name}:ingress-proxy")
+    await ops_test.model.wait_for_idle(apps=apps, status=ActiveStatus.name, timeout=60 * 5)
+    await ops_test.model.add_relation(f"{app_name}:ingress", nginx_integrator_app.name)
+    await ops_test.model.wait_for_idle(apps=apps, status=ActiveStatus.name, timeout=60 * 5)
+
+    assert ops_test.model.applications[app_name].units[0].workload_status == ActiveStatus.name
+    assert (
+        ops_test.model.applications[hello_kubecon_app_name].units[0].workload_status
+        == ActiveStatus.name
+    )
+
     yield application
+
+
+@pytest_asyncio.fixture(scope="module")
+async def ip_address_list(ops_test: OpsTest, app: Application, nginx_integrator_app: Application):
+    """Get unit IP address from workload message.
+    Example: Ingress IP(s): 127.0.0.1, Service IP(s): 10.152.183.84
+    """
+    # Reduce the update_status frequency until the cluster is deployed
+    async with ops_test.fast_forward():
+        await ops_test.model.block_until(
+            lambda: "Ingress IP(s)" in nginx_integrator_app.units[0].workload_status_message,
+            timeout=100,
+        )
+    nginx_integrator_unit = nginx_integrator_app.units[0]
+    status_message = nginx_integrator_unit.workload_status_message
+    ip_regex = r"[0-9]+(?:\.[0-9]+){3}"
```

Comment:
No possibility of IPv6 address that won't match this regex ?

Reply:
I doubt it can match any IPv6 address, given there are no a-f symbols and the limitation of 4 fields.

Reply:
Yeap that's what I meant, my question rephrased is, is there any chance that instead of an IPv4 this address is an IPv6 which won't be matched by this regex ?

Reply:
I'll create a backlog story specifically for that just in case.